### DISCUSSION TITLE
How can I append a new JSONB entry into a JSONB cell?

### DISCUSSION THREAD

**Discussion by sbutler-gh** [supabase/supabase#1674] [Questions] [answered]
In my database, I have a `PROJECTS` table, with a column `subscribers` for individuals who are subscribed to specific projects (rows)

Currently, the data in the `subscribers` column is just a JSONB entry of an email address.

How can I add new email addresses to the `subscribers` cell of a given row?

For example, **Project A** has `subscribers` data equal to `["amy@first.com"]`

Then, another person (bob@second.com) subscribes to Project A.

The 'subscribers' data for Project A should now be something like `["amy@first.com", "bob@second.com"]`

Is there any way I can do this with the Supabase methods out of the box?  Or would I need to do an RPC? (If RPC is needed, could you please advise on the correct format for this procedure?)

**Comment by soedirgo** [comment, marked as answer]
Hey @sbutler-gh, you can try using the JSONB version in https://github.com/supabase/supabase/discussions/1570#discussioncomment-742428

**Reply by sbutler-gh** [reply]
Thank you @soedirgo !

This is the function I used in the end to create the RPC:
```
create function append_item (row_id int, new_item jsonb) 
returns void as 
$$
  update table
  set value = coalesce(value, '[]'::jsonb) || to_jsonb(new_item)
  where id = row_id
  returning *;
$$ 
language sql;
```

And then inserting that into application with:

```
let new_item = [value to be added to array]
let row_id = [the id/key of the row, which contains the array that should be updated]

    let { data, error } = await supabase
    .rpc('append_item', {
      new_item, 
      row_id
    })
```
